### PR TITLE
Add getpass(), getpass_r() and getpassfd() from NetBSD

### PR DESCRIPTION
**Pull request by j-piecuch**
`getpassfd()` has been modified slightly so that it doesn't use `poll()`. As a consequence, it ignores the `tout` argument.

If you want to test it, the `getpass-test` branch on my fork has a `getpass` program which shows that it works.

### CODE REVIEW

File: lib/libc/gen/getpass.c [hunk 1]
```diff
@@ -1,6 +1,332 @@
...
+      return e;
+  }
+
+  if ((rv = getpass_r(prompt, buf, (size_t)bufsiz)) == NULL)
+    return e;
+
+  return rv;
+}
+
+#ifdef TEST
```
Comment (franciscozdo) [inline lib/libc/gen/getpass.c:325]: Do we run this?

**Review by franciscozdo — COMMENTED** [review]
Looks good, but I think you have some unused code.

File: lib/libc/gen/getpass.c [hunk 2]
```diff
@@ -1,6 +1,332 @@
...
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+#include <sys/cdefs.h>
+#if defined(LIBC_SCCS) && !defined(lint)
+__RCSID("$NetBSD: getpass.c,v 1.30 2016/01/31 23:41:38 christos Exp $");
```
[the remark this answers is not in the document]
Comment (j-piecuch) [reply lib/libc/gen/getpass.c]: Done.

File: lib/libc/gen/getpass.c (hunk 1 above)
Comment (j-piecuch) [reply lib/libc/gen/getpass.c]: No. Removed.

**Review by franciscozdo — APPROVED** [review]
LGTM

**Pull request merged**